### DISCUSSION TITLE
I am a bit confused about installing Poetry

### DISCUSSION THREAD

**Discussion by ghost** [python-poetry/poetry#9838] [General]
So from my understanding, which is most likely wrong, you first have to navigate to your project, create a virtual environment inside the project directory, and then you run `pipx install poetry`, or `pip install poetry` (I would personally prefer using `pip`). And all this results in Poetry getting installed onto your project, where you can then manage dependencies and package your python script.

However, the documentation says "Poetry should always be installed in a dedicated virtual environment to isolate it from the rest of your system. It should in no case be installed in the environment of the project that is to be managed by Poetry", so I am a bit confused as to where I should install Poetry.

Can anyone give some recommendations please? I am completely new to Poetry, so forgive me if I make any mistakes or anger you. Thanks!

**Comment by radoering** [comment]
> you first have to navigate to your project, create a virtual environment inside the project directory, and then you run `pipx install poetry`

No, you do not need a project to install Poetry. pipx will automatically install Poetry in a dedicated environment, which is separated from your project's environment.

> or pip install poetry (I would personally prefer using pip)

No, you should not install Poetry into your project's environment. If you want to use pip instead of pipx, you have to do some manual steps. See https://python-poetry.org/docs/#installing-manually

You always need one more environment than projects because you need an additional environment for Poetry itself. (You must not activate Poetry's environment, you can just use the Poetry executable.) You can create your environments for your projects manually (so Poetry will use them if they are activated) or let Poetry create them automatically (either in a central directory or in a .venv directory in each of your projects).

**Reply by ghost** [reply]
> 

> > you first have to navigate to your project, create a virtual environment inside the project directory, and then you run `pipx install poetry`
> 
> No, you do not need a project to install Poetry. pipx will automatically install Poetry in a dedicated environment, which is separated from your project's environment.
> 
> > or pip install poetry (I would personally prefer using pip)
> 
> No, you should not install Poetry into your project's environment. If you want to use pip instead of pipx, you have to do some manual steps. See https://python-poetry.org/docs/#installing-manually
> 
> You always need one more environment than projects because you need an additional environment for Poetry itself. (You must not activate Poetry's environment, you can just use the Poetry executable.) You can create your environments for your projects manually (so Poetry will use them if they are activated) or let Poetry create them automatically (either in a central directory or in a .venv directory in each of your projects).

Alright, so I should create a virtual environment at the root of my computer, and then install pipx, and then Install Poetry?

Then every time I make a new python project, poetry will always be available.
